calico_mesos/calico_mesos.py

```diff
@@ -259,13 +264,16 @@ def _isolate(hostname, ns_pid, container_id, ipv4_addrs, ipv6_addrs, profiles, l
                                    ip_list=ipv4_addrs)
 
     # Create any profiles in etcd that do not already exist
-    if profiles == []:
-        profiles = ["default"]
     _log.info("Assigning Profiles: %s" % profiles)
     for profile in profiles:
-        # Create profile with default rules, if it does not exist
+        profile = "ng_%s" % profile
         if not datastore.profile_exists(profile):
-            _create_profile_with_default_mesos_rules(profile)
+            _create_profile_for_netgroup(profile)
+
+    _log.info("Assigning Default Host Profile: %s" % profiles)
+    default_profile_name = "default_%s" % hostname
+    if not datastore.profile_exists(default_profile_name):
+        _create_profile_for_host_communication(default_profile_name)
```

Comment:
This code leaves `profiles` unmodified!  You need to

1.  use the new `ng_` names
2.  prepend the `default_profile_name`